client/extensions/woocommerce/app/products/product-create.js

```diff
@@ -127,7 +123,9 @@ class ProductCreate extends React.Component {
 
 function mapStateToProps( state ) {
 	const site = getSelectedSiteWithFallback( state );
-	const product = getCurrentlyEditingProduct( state );
+	const productId = getCurrentlyEditingId( state, site.id );
```

Comment:
This returns an error when `site` isn't loaded yet (you can see this when initial state rehydration is skipped). 

<img width="1003" alt="screen shot 2017-07-04 at 2 46 36 pm" src="https://user-images.githubusercontent.com/689165/27843366-a2c644d2-60c7-11e7-9e4c-b2c7b03fce75.png">